### ISSUE TITLE
Be warned of missing sphinx version on Heroku

### ISSUE THREAD

**Issue by pedrocunha** [flying-sphinx/flying-sphinx#17]
Hi,

I've just ran against an issue on Heroku where the sphinx version `2.2.6` (present in `config/thinking_sphinx.yml`) was not present on flying sphinx servers and it fallback to 1.10-beta. Unfortunately only got this output too late (after the rebuild task was over), and I experienced a few minutes downtime in production regarding this issue.

In best world scenario I would love to:
- Have a task that outputs Sphinx version on FlyingSphinx servers
- Have FlyingSphinx somehow warning the version mismatch / not found - preferentially whenever any rake task in the `fs` domain is invoked.
- Have the FlyingSphinx admin (via heroku) also reporting current Sphinx version
- Have FlyingSphinx still warning version mismatch but falling back to latest minor version available for the major version specified.

Is there any chance we could improve this somehow ?

**Comment by pat** [comment]
(Sorry for the slow reply).

All of this sounds great - I've been pondering how to implement at least part of it - I think a warning or error if the version doesn't exist would be ideal. The available versions are covered here: http://support.flying-sphinx.com/kb/configuration/setting-your-sphinx-version - but that could be advertised far better in the documentation.

And certainly displaying the version in the dashboard would be useful too.

All in all, this is now on my radar, will keep you posted when any/all of it is implemented.

**Comment by pedrocunha** [comment]
No worries. That's great! This would be really helpful.

**Comment by pat** [comment]
Three years later… :S

This issue got pondered about, and then forgotten about. Initially I couldn't see an elegant way of going about it, but just now I've had inspiration for a solid approach and I'm really not sure what stopped me in the first place.

So, I've just rolled out a specific API endpoint to confirm a given engine/version is available on the servers, and if it returns a 404, then `flying-sphinx` will raise an exception immediately, before uploading any configuration. It's all part of the `ts:configure` rake task, and is also invoked via `ts:index` and `ts:rebuild`. I've just released v2.1.0 with this patch (the v2.x releases require TS v4.x though).

Again, thank you so much for the suggestion (whether or not you're still using Flying Sphinx!), and I'm very sorry it took so long to get around to sorting it out.
